### PR TITLE
add pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by bissenbay**
Related-To: https://github.com/thoth-station/invectio/issues/82

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    hooks:
+      - id: black
+
+  - repo: https://gitlab.com/PyCQA/flake8
+    rev: 3.8.3
+    hooks:
+      - id: flake8
+        additional_dependencies: ['pep8-naming']
+        # Ignore all format-related checks as Black takes care of those.
+        args: ['--ignore', 'E2,W5,F401,N802', '--select', 'E,W,F,N', '--max-line-length=136']
```
Comment (bissenbay) [inline .pre-commit-config.yaml:51]: @fridex I ignored the following error codes `F401`, `N802` and removed the below part from `pre-commit-config.yaml`
```
- repo: git://github.com/pycqa/pydocstyle.git
    rev: 5.0.2
    hooks:
      - id: pydocstyle
```
Is it OK?
They gave a lot of errors like `missing docstrings` and `lowercasing most of the function names`. I can modify them, I'm afraid they maybe used somewhere and changing them might break something, what do you think I should do `ignore` or `work on those errors`?

Comment (fridex) [reply .pre-commit-config.yaml:51]: F401 and N802 sound reasonable to have to me. Any specific reason why they have been removed?

Open a PR with a standard pre-commit config we have, I can take a look and add docstrings if it requires my action.

Comment (bissenbay) [reply .pre-commit-config.yaml]: @fridex , I added standard `pre-commit-config.yaml` from https://github.com/thoth-station/template-project, https://github.com/thoth-station/invectio/pull/93/commits/998dd10774ea248803b8aff629dd955137466509

**Comment by fridex** [comment]
> Pre-Commit Test failed! Click here
> ```
> [INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
> [INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
> [INFO] Initializing environment for git://github.com/pycqa/pydocstyle.git.
> [INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
> [INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy.
> [INFO] Initializing environment for https://github.com/psf/black.
> [INFO] Initializing environment for https://gitlab.com/PyCQA/flake8.
> [INFO] Initializing environment for https://gitlab.com/PyCQA/flake8:pep8-naming.
> [INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for git://github.com/pycqa/pydocstyle.git.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://github.com/pre-commit/mirrors-mypy.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://github.com/psf/black.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://gitlab.com/PyCQA/flake8.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> Tabs remover.............................................................Passed
> Trim Trailing Whitespace.................................................Passed
> Check for merge conflicts................................................Passed
> Fix End of Files.........................................................Failed
> - hook id: end-of-file-fixer
> - exit code: 1
> - files were modified by this hook
> 
> Fixing MANIFEST.in

This one is easily fixable.

> Tests should end in _test.py.............................................Failed
> - hook id: name-tests-test
> - exit code: 1
> 
> tests/data/project_dir/proj/model.py does not match pattern ".*_test.py"
> tests/data/app_9.py does not match pattern ".*_test.py"
> tests/data/app_7.py does not match pattern ".*_test.py"
> tests/data/empty.py does not match pattern ".*_test.py"
> tests/data/app_8.py does not match pattern ".*_test.py"
> tests/data/project_dir/proj/utils.py does not match pattern ".*_test.py"
> tests/data/app_2.py does not match pattern ".*_test.py"
> tests/data/app_5.py does not match pattern ".*_test.py"
> tests/data/app_3.py does not match pattern ".*_test.py"
> tests/data/app_6.py does not match pattern ".*_test.py"
> tests/data/project_dir/main.py does not match pattern ".*_test.py"
> tests/data/lstm.py does not match pattern ".*_test.py"
> tests/data/app_4.py does not match pattern ".*_test.py"
> tests/data/app_1.py does not match pattern ".*_test.py"
> tests/test_lib.py does not match pattern ".*_test.py"

You can ignore these files - they are used as inputs in the testsuite and do not implement any testing functionality.

> Check for added large files..............................................Passed
> Check for byte-order marker..............................................Passed
> Check for case conflicts.................................................Passed
> Check docstring is first.................................................Failed
> - hook id: check-docstring-first
> - exit code: 1
> 
> tests/data/lstm.py:24 Multiple module docstrings (first docstring on line 1).
> 
> Check JSON...............................................................Passed
> Check for broken symlinks................................................Passed
> Detect Private Key.......................................................Passed
> Check python ast.........................................................Passed
> Debug Statements (Python)................................................Passed
> pydocstyle...............................................................Failed
> - hook id: pydocstyle
> - exit code: 1
> 
> setup.py:1 at module level:
>         D100: Missing docstring in public module
> setup.py:11 in public class `Test`:
>         D101: Missing docstring in public class
> setup.py:14 in public method `initialize_options`:
>         D102: Missing docstring in public method
> setup.py:29 in public method `finalize_options`:
>         D102: Missing docstring in public method
> setup.py:34 in public method `run_tests`:
>         D102: Missing docstring in public method
> setup.py:40 in public function `get_version`:
>         D103: Missing docstring in public function
> tests/data/app_5.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/project_dir/proj/model.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/project_dir/proj/model.py:6 in public function `get_model`:
>         D103: Missing docstring in public function
> tests/data/app_3.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/app_6.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/app_2.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/app_7.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/app_7.py:5 in public function `now`:
>         D103: Missing docstring in public function
> tests/test_lib.py:1 at module level:
>         D100: Missing docstring in public module
> tests/test_lib.py:39 in public method `test_version`:
>         D102: Missing docstring in public method
> tests/test_lib.py:45 in public method `test_empty`:
>         D102: Missing docstring in public method
> tests/test_lib.py:51 in public method `test_no_files`:
>         D102: Missing docstring in public method
> tests/test_lib.py:56 in public method `test_app_1`:
>         D102: Missing docstring in public method
> tests/test_lib.py:64 in public method `test_app_2`:
>         D102: Missing docstring in public method
> tests/test_lib.py:72 in public method `test_app_3`:
>         D102: Missing docstring in public method
> tests/test_lib.py:80 in public method `test_app_4`:
>         D102: Missing docstring in public method
> tests/test_lib.py:88 in public method `test_app_5`:
>         D102: Missing docstring in public method
> tests/test_lib.py:96 in public method `test_app_6`:
>         D102: Missing docstring in public method
> tests/test_lib.py:104 in public method `test_lstm`:
>         D102: Missing docstring in public method
> tests/test_lib.py:133 in public method `test_project_dir`:
>         D102: Missing docstring in public method
> tests/test_lib.py:155 in public method `test_standard_imports_detection`:
>         D102: Missing docstring in public method
> tests/test_lib.py:173 in public method `test_without_builtin_and_standard_imports`:
>         D102: Missing docstring in public method
> tests/test_lib.py:187 in public method `test_without_builtin_imports`:
>         D102: Missing docstring in public method
> tests/test_lib.py:240 in public method `test_version`:
>         D102: Missing docstring in public method
> tests/test_lib.py:246 in public method `test_empty`:
>         D102: Missing docstring in public method
> tests/test_lib.py:252 in public method `test_no_files`:
>         D102: Missing docstring in public method
> tests/test_lib.py:257 in public method `test_app9`:
>         D102: Missing docstring in public method
> tests/test_lib.py:275 in public function `test_get_standard_imports`:
>         D103: Missing docstring in public function
> tests/data/app_8.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/app_8.py:8 in public function `signal_handler`:
>         D103: Missing docstring in public function
> tests/data/app_9.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/app_9.py:15 in public class `SomeClass`:
>         D101: Missing docstring in public class
> tests/data/app_9.py:16 in public nested class `NestedClass`:
>         D106: Missing docstring in public nested class
> tests/data/app_9.py:19 in public method `method`:
>         D102: Missing docstring in public method
> tests/data/app_9.py:23 in public function `async_signal_handler`:
>         D103: Missing docstring in public function
> tests/data/app_9.py:27 in public function `signal_handler`:
>         D103: Missing docstring in public function
> tests/data/project_dir/main.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/project_dir/main.py:8 in public function `get_model`:
>         D103: Missing docstring in public function
> tests/data/app_4.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/empty.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/project_dir/proj/__init__.py:1 at module level:
>         D104: Missing docstring in public package
> tests/data/lstm.py:1 at module level:
>         D210: No whitespaces allowed surrounding docstring text
> tests/data/lstm.py:51 in public function `RNN`:
>         D103: Missing docstring in public function
> tests/data/app_1.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/project_dir/proj/utils.py:1 at module level:
>         D100: Missing docstring in public module
> tests/data/project_dir/proj/utils.py:4 in public function `now`:
>         D103: Missing docstring in public function

You can ignore docstrings in the test files. Others can be added.

> Check Toml...............................................................Passed
> Check Yaml...............................................................Passed
> Fix End of Files.........................................................Passed
> Trim Trailing Whitespace.................................................Passed
> mypy.....................................................................Failed
> - hook id: mypy
> - exit code: 1
> 
> invectio/lib.py:363: error: Argument 1 to "visit" of "NodeVisitor" has incompatible type "object"; expected "AST"
> invectio/lib.py:398: error: Argument 1 to "visit" of "InvectioSymbolsProvidedVisitor" has incompatible type "object"; expected Module
> invectio/cli.py:107: error: Argument 1 to "gather_library_usage" has incompatible type "Optional[str]"; expected "str"
> invectio/cli.py:137: error: Argument 1 to "gather_symbols_provided" has incompatible type "Optional[str]"; expected "str"
> Found 4 errors in 2 files (checked 3 source files)

This needs to be fixed.

> black....................................................................Passed
> flake8...................................................................Failed
> - hook id: flake8
> - exit code: 1
> 
> setup.py:6:2: N812 lowercase 'test' imported as non lowercase 'TestCommand'
> setup.py:54:121: E501 line too long (136 > 120 characters)
> tests/data/app_7.py:6:5: F841 local variable 'q' is assigned to but never used
> tests/data/project_dir/main.py:7:1: F811 redefinition of unused 'get_model' from line 2
> invectio/lib.py:53:10: N802 function name 'visit_Import' should be lowercase
> invectio/lib.py:68:10: N802 function name 'visit_ImportFrom' should be lowercase
> invectio/lib.py:99:10: N802 function name 'visit_Attribute' should be lowercase
> invectio/lib.py:116:10: N802 function name 'visit_Name' should be lowercase
> invectio/lib.py:169:10: N802 function name 'visit_FunctionDef' should be lowercase
> invectio/lib.py:183:10: N802 function name 'visit_AsyncFunctionDef' should be lowercase
> invectio/lib.py:197:10: N802 function name 'visit_ClassDef' should be lowercase
> invectio/lib.py:211:10: N802 function name 'visit_Assign' should be lowercase
> invectio/lib.py:255:10: N802 function name 'visit_AnnAssign' should be lowercase
> tests/data/project_dir/proj/__init__.py:1:1: F401 '.model.get_model' imported but unused
> tests/data/lstm.py:51:6: N802 function name 'RNN' should be lowercase
> tests/data/app_1.py:1:34: E741 ambiguous variable name 'l'
> invectio/__init__.py:26:1: F401 '.lib.gather_library_usage' imported but unused
> invectio/__init__.py:27:1: F401 '.lib.gather_symbols_provided' imported but unused
> invectio/__init__.py:28:1: F401 '.lib.get_standard_imports' imported but unused
> ```

Sources in tests can be ignored. Regarding the function naming convention - it is expected by the ast module so we cannot rename methods to lowercase. They can be ignored on source code level using directives.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      - id: check-symlinks
+      - id: detect-private-key
+      - id: check-ast
+      - id: debug-statements
+
+  - repo: git://github.com/pycqa/pydocstyle.git
+    rev: 5.1.1
+    hooks:
+      - id: pydocstyle
+        exclude: '^tests*'
```
Comment (bissenbay) [inline .pre-commit-config.yaml:29]: My intention here to exclude `tests` directory from checking by `pydocstyle`, not sure whether it is the correct pattering matching?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  #   rev: '0.39'
+  #   hooks:
+  #     - id: check-manifest
+
+  - repo: https://gitlab.com/PyCQA/flake8
+    rev: '3.7.8'
+    hooks:
+      - id: flake8
+        additional_dependencies: ['pep8-naming']
+        # Ignore all format-related checks as Black takes care of those.
```
Comment (bissenbay) [inline .pre-commit-config.yaml:62]: I wanted to add `exclude: '^tests*'` here as well, so `flake8` is not checked agains `tests` directory, but instead I went with ignoring inline on lines https://github.com/thoth-station/invectio/pull/93/files#diff-c520182137ed753e8c43f2e27dbd06c912c5367100734ff6c9f635146b0ec929R1, https://github.com/thoth-station/invectio/pull/93/files?file-filters%5B%5D=.py&file-filters%5B%5D=.yaml#diff-7eacf9ce8f12f34111bc067f2516c9c1c30b34fad589938719cfd3fd764f308bR6, https://github.com/thoth-station/invectio/pull/93/files?file-filters%5B%5D=.py&file-filters%5B%5D=.yaml#diff-b34ca95fc6e4f3df60e3146e057a4e59541ef2386d9cfbe764456981b86bbd07R49

File: invectio/__init__.py [hunk 4]
```diff
@@ -17,12 +17,18 @@
...
 __version__ = "0.1.0"
 __author__ = "Fridolin Pokorny"
 __email__ = "fridex.devel@gmail.com"
 __title__ = "invectio"
 
 
-from .lib import gather_library_usage
-from .lib import gather_symbols_provided
-from .lib import get_standard_imports
+__all__ = [
```
Comment (bissenbay) [inline invectio/__init__.py:30]: added `__all__` to skip 
```
invectio/init.py:26:1: F401 '.lib.gather_library_usage' imported but unused
invectio/init.py:27:1: F401 '.lib.gather_symbols_provided' imported but unused
invectio/init.py:28:1: F401 '.lib.get_standard_imports' imported but unused
```

File: tests/data/lstm_test.py [hunk 5]
```diff
@@ -21,11 +21,9 @@
 
 mnist = input_data.read_data_sets("/tmp/data/", one_hot=True)
 
-"""
-To classify images using a recurrent neural network, we consider every image
-row as a sequence of pixels. Because MNIST image shape is 28*28px, we will then
-handle 28 sequences of 28 steps for every sample.
-"""
+# To classify images using a recurrent neural network, we consider every image
```
Comment (bissenbay) [inline tests/data/lstm_test.py:24]: Modified as comments to skip `tests/data/lstm.py:24 Multiple module docstrings (first docstring on line 1).`

File: tests/lib_test.py [hunk 6]
```diff
@@ -37,13 +37,13 @@ class TestLibraryUsage(InvectioTestBase):
     """A base class for library usage related tests."""
 
     def test_version(self) -> None:
-        file_path = self._get_test_path("empty.py")
+        file_path = self._get_test_path("empty_test.py")
```
Comment (bissenbay) [inline tests/lib_test.py:40]: renamed all files inside `test` to end with `_test.py`.

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        args: ['--ignore', 'E2,W5', '--select', 'E,W,F,N', '--max-line-length=136']
```
Comment (bissenbay) [inline .pre-commit-config.yaml:63]: changed `'--max-line-length=136'` to allow more characters for the description: https://github.com/thoth-station/invectio/blob/master/setup.py#L54

**Review by bissenbay — COMMENTED** [review]
@fridex PTAL, let me know if I need to change anything

**Comment by bissenbay** [comment]
> Pre-Commit Test failed! Click here
> ```
> [INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
> [INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
> [INFO] Initializing environment for git://github.com/pycqa/pydocstyle.git.
> [INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
> [INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy.
> [INFO] Initializing environment for https://github.com/psf/black.
> [INFO] Initializing environment for https://gitlab.com/PyCQA/flake8.
> [INFO] Initializing environment for https://gitlab.com/PyCQA/flake8:pep8-naming.
> [INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for git://github.com/pycqa/pydocstyle.git.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://github.com/pre-commit/mirrors-mypy.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://github.com/psf/black.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> [INFO] Installing environment for https://gitlab.com/PyCQA/flake8.
> [INFO] Once installed this environment will be reused.
> [INFO] This may take a few minutes...
> Tabs remover.............................................................Passed
> Trim Trailing Whitespace.................................................Passed
> Check for merge conflicts................................................Passed
> Fix End of Files.........................................................Passed
> Tests should end in _test.py.............................................Passed
> Check for added large files..............................................Passed
> Check for byte-order marker..............................................Passed
> Check for case conflicts.................................................Passed
> Check docstring is first.................................................Passed
> Check JSON...............................................................Passed
> Check for broken symlinks................................................Passed
> Detect Private Key.......................................................Passed
> Check python ast.........................................................Passed
> Debug Statements (Python)................................................Passed
> pydocstyle...............................................................Passed
> Check Toml...............................................................Passed
> Check Yaml...............................................................Passed
> Fix End of Files.........................................................Passed
> Trim Trailing Whitespace.................................................Passed
> mypy.....................................................................Passed
> black....................................................................Passed
> flake8...................................................................Failed
> - hook id: flake8
> - exit code: 1
> 
> tests/data/project_dir/main_test.py:7:1: F811 redefinition of unused 'get_model' from line 2
> ```

@harshad16 Here's another example where `pre-commit` passes locally, but fails on `ci run`.

**Comment by bissenbay** [comment]
changes in https://github.com/thoth-station/invectio/pull/93/commits/d630211a570c913923973f5548a7d96578051b2a to prevent https://github.com/thoth-station/invectio/pull/93#issuecomment-726247776

**Comment by bissenbay** [comment]
In https://github.com/thoth-station/invectio/pull/93/commits/eaab01c0a8b8c238f7d3a96ba73bc0bea7b5235f, the below `pytest` error is fixed:
```
________________________ TestSymbolsProvided.test_app9 _________________________

self = <lib_test.TestSymbolsProvided object at 0x7f77d30f9588>

    def test_app9(self) -> None:
        file_path = self._get_test_path("app_9_test.py")
        result = gather_symbols_provided(file_path)
        assert "report" in result
        assert str(file_path) in result["report"]
>       assert set(result["report"][str(file_path)]) == {
            "tests.data.app_9.A",
            "tests.data.app_9.B",
            "tests.data.app_9.GLOBAL_VAL",
            "tests.data.app_9.SomeClass",
            "tests.data.app_9.X",
            "tests.data.app_9.Y",
            "tests.data.app_9.async_signal_handler",
            "tests.data.app_9.b",
            "tests.data.app_9.signal_handler",
        }
E       AssertionError: assert {'tests.data.app_9_test.A',\n 'tests.data.app_9_test.B',\n 'tests.data.app_9_test.GLOBAL_VAL',\n 'tests.data.app_9_test.SomeClass',\n 'tests.data.app_9_test.X',\n 'tests.data.app_9_test.Y',\n 'tests.data.app_9_test.async_signal_handler',\n 'tests.data.app_9_test.signal_handler'} == {'tests.data.app_9.A',\n 'tests.data.app_9.B',\n 'tests.data.app_9.GLOBAL_VAL',\n 'tests.data.app_9.SomeClass',\n 'tests.data.app_9.X',\n 'tests.data.app_9.Y',\n 'tests.data.app_9.async_signal_handler',\n 'tests.data.app_9.b',\n 'tests.data.app_9.signal_handler'}
E         Extra items in the left set:
E         'tests.data.app_9_test.signal_handler'
E         'tests.data.app_9_test.A'
E         'tests.data.app_9_test.async_signal_handler'
E         'tests.data.app_9_test.Y'
E         'tests.data.app_9_test.SomeClass'
E         'tests.data.app_9_test.X'
E         'tests.data.app_9_test.GLOBAL_VAL'
E         'tests.data.app_9_test.B'
E         Extra items in the right set:
E         'tests.data.app_9.GLOBAL_VAL'
E         'tests.data.app_9.Y'
E         'tests.data.app_9.X'
E         'tests.data.app_9.signal_handler'
E         'tests.data.app_9.B'
E         'tests.data.app_9.async_signal_handler'
E         'tests.data.app_9.A'
E         'tests.data.app_9.b'
E         'tests.data.app_9.SomeClass'
E         Full diff:
E           {
E         -  'tests.data.app_9.A',
E         +  'tests.data.app_9_test.A',
E         ?                   +++++
E         -  'tests.data.app_9.B',
E         +  'tests.data.app_9_test.B',
E         ?                   +++++
E         -  'tests.data.app_9.GLOBAL_VAL',
E         +  'tests.data.app_9_test.GLOBAL_VAL',
E         ?                   +++++
E         -  'tests.data.app_9.SomeClass',
E         +  'tests.data.app_9_test.SomeClass',
E         ?                   +++++
E         -  'tests.data.app_9.X',
E         +  'tests.data.app_9_test.X',
E         ?                   +++++
E         -  'tests.data.app_9.Y',
E         +  'tests.data.app_9_test.Y',
E         ?                   +++++
E         -  'tests.data.app_9.async_signal_handler',
E         +  'tests.data.app_9_test.async_signal_handler',
E         ?                   +++++
E         -  'tests.data.app_9.b',
E         -  'tests.data.app_9.signal_handler',
E         +  'tests.data.app_9_test.signal_handler',
E         ?                   +++++
E           }

file_path  = 'tests/data/app_9_test.py'
result     = {'report': {'tests/data/app_9_test.py': ['tests.data.app_9_test.A',
                                         'tests.data.app_9_test.B',
                                         'tests.data.app_9_test.GLOBAL_VAL',
                                         'tests.data.app_9_test.SomeClass',
                                         'tests.data.app_9_test.X',
                                         'tests.data.app_9_test.Y',
                                         'tests.data.app_9_test.async_signal_handler',
                                         'tests.data.app_9_test.signal_handler']},
 'version': '0.1.0'}
self       = <lib_test.TestSymbolsProvided object at 0x7f77d30f9588>

tests/lib_test.py:262: AssertionError
```

File: tests/lib_test.py [hunk 8]
```diff
@@ -255,20 +255,19 @@ def test_no_files(self) -> None:
...
         assert str(file_path) in result["report"]
         assert set(result["report"][str(file_path)]) == {
-            "tests.data.app_9.A",
-            "tests.data.app_9.B",
-            "tests.data.app_9.GLOBAL_VAL",
-            "tests.data.app_9.SomeClass",
-            "tests.data.app_9.X",
-            "tests.data.app_9.Y",
-            "tests.data.app_9.async_signal_handler",
-            "tests.data.app_9.b",
```
Comment (bissenbay) [inline tests/lib_test.py:270]: @fridex this line was failing `pytest`, so I deleted, WDYT, is it OK?

**Comment by bissenbay** [comment]
@fridex PTAL

File: tests/lib_test.py (hunk 8 above)
Comment (fridex) [reply tests/lib_test.py:270]: This looks suspicious. Why was that? Was it failing even before?

File: tests/data/lstm_test.py (hunk 5 above)
Comment (fridex) [reply tests/data/lstm_test.py:24]: OK :+1:

File: invectio/lib.py [hunk 9]
```diff
@@ -252,7 +254,7 @@ def _maybe_add(node: ast.AST):
                         f"Failed to parse assign statement in {self.file_name}"
                     )
 
-    def visit_AnnAssign(self, ann_assign: ast.AugAssign) -> None:
```
Comment (fridex) [inline invectio/lib.py:255]: This looks like a logic change. I think it should be kept named `visit_AnnAssign` to visit AnnAssign.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (fridex) [reply .pre-commit-config.yaml:29]: Looks like it could work :+1:

**Review by fridex — COMMENTED** [review]
Few minor comments, otherwise looks good to me :+1:

File: invectio/lib.py (hunk 9 above)
Comment (bissenbay) [reply invectio/lib.py:255]: thanks, fixed https://github.com/thoth-station/invectio/pull/93/commits/0a2d92553caf937cd2bb422c688c4714b3cfb074

**Comment by bissenbay** [comment]
@fridex PTAL!

File: tests/lib_test.py (hunk 8 above)
Comment (fridex) [reply tests/lib_test.py:270]: Looks like fixed now.

**Review by fridex — APPROVED** [review]
looks good to me :+1: nice job! :100:

**Pull request merged**